WooCommerce/Classes/ViewRelated/Dashboard/Settings/Beta features/BetaFeaturesViewController.swift

```diff
@@ -0,0 +1,200 @@
+import Storage
+import UIKit
+import Yosemite
+
+/// Contains UI for Beta features that can be turned on and off.
+///
+class BetaFeaturesViewController: UIViewController {
+
+    /// Main TableView
+    ///
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero, style: .grouped)
+        return tableView
+    }()
+
+    /// Table Sections to be rendered
+    ///
+    private var sections = [Section]()
+
+    private let siteID: Int
+
+    init(siteID: Int) {
+        self.siteID = siteID
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Overridden Methods
+    //
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigationBar()
+        configureMainView()
+        configureSections()
+        configureTableView()
+        registerTableViewCells()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension BetaFeaturesViewController {
+
+    /// Set the title.
+    ///
+    func configureNavigationBar() {
+        title = NSLocalizedString("Beta Features", comment: "Beta features navigation title")
+    }
+
+    /// Apply Woo styles.
+    ///
+    func configureMainView() {
+        view.backgroundColor = StyleManager.tableViewBackgroundColor
+    }
+
+    /// Configure common table properties.
+    ///
+    func configureTableView() {
+        view.addSubview(tableView)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        view.pinSubviewToAllEdges(tableView)
+
+        tableView.dataSource = self
+
+        tableView.estimatedRowHeight = Constants.rowHeight
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = StyleManager.tableViewBackgroundColor
+    }
+
+    /// Configure sections for table view.
+    ///
+    func configureSections() {
+        sections = [
+            Section(rows: [.statsVersionSwitch,
+                           .statsVersionDescription])
+        ]
+    }
+
+    /// Register table cells.
+    ///
+    func registerTableViewCells() {
+        for row in Row.allCases {
+            tableView.register(row.type.loadNib(), forCellReuseIdentifier: row.reuseIdentifier)
+        }
+    }
+
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        guard type(of: cell) == row.type else {
+            fatalError()
```

Comment:
Is this a runtime crash? If it is, maybe throwing an assertion might be slightly preferable, as it still warns us devs that we did not register the cell properly?